content/en/docs/installing-on-linux/public-cloud/install-ks-on-azure-vms.md

```diff
@@ -0,0 +1,208 @@
+
+![Architecture](/images/docs/aks/Azure-architecture.png)
+
+Those VMs will be attached to a LoadBalancer, there are two different rules are per-defined in the LB:
+
+- **Inbound NAT**: ssh port will be mapped for each machine, so we can easily manage VMs.
+- **Load Balancing**: http and https ports will be mapped to Node pools by default, we can add other ports on demand.
+
+| Service | Protocol | Rule | Backend Port | Frontend Port/Ports | Pools |
+|---|---|---|---|---|---|
+| ssh | TCP | Inbound NAT | 22 |50200, 50201,50202, 50100~50199| Master, Node |
+| apiserver | TCP | Load Balancing | 6443 | 6443 | Master |
+| ks-console | TCP | Load Balancing | 30880 | 30880 | Master |
+| http | TCP | Load Balancing | 80 | 80 | Node |
+| https | TCP | Load Balancing | 443 | 443 | Node |
+
+## Deploy HA Cluster Infrastructrue
+
+You don't have to create those resources one by one with Wizards. Following the best practice of **infrastructure as code** on Azure, all resources in the architecture are already defined as ARM templates.
+
+### Start to deploy with one click
+
+Click the *Deploy* button below, you will be redirected to Azure and asked to fill in deployment parameters.
+
+[![Deploy to Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FRolandMa1986%2Fazurek8s%2Fmaster%2Fazuredeploy.json) [![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FRolandMa1986%2Fazurek8s%2Fmaster%2Fazuredeploy.json)
+
+### Change template parameters
+
+Only a few parameter need to be changed.
+
+- Choose the *Create new* link under the Resource group and fill in a Name such as "KubeSphereVMRG".
+- Fill in the Admin Username.
+- Copy your public ssh key and fill in the Admin Key. Or create new one with *ssh-keygen*.
+
+> Password authentication is restriced in the Linux configuration, only SSH accept.
+
+Click the *Purchase* button in the bottom when you ready to continue.
+
+### Review Azure Resources in the Portal
+
+Once the deployment success, you can find all the resources you need in the KubeSphereVMRG resource group. Take your time and check them one by one if you are new to Azure. Then find the public IP of LB and private IP addresses of the VMs. You need them in the next step.
+
+![New Created Resources](/images/docs/aks/azure-vm-all-resources.png)
+
+## Deploy Kubernetes and Kubesphere
+
+For the following commands, you can execute them on your laptop or SSH to one of the Master VMs. There are files will be downloaded to local and disturbed to each VM during the installation. The installation will be much faster when you use **kk** in the Intranet than the Internet.
```

Comment:
you can execute the following command on your laptop or SSH to one of the Master VMs.